lib/rules/no-useless-assignment.js

```diff
@@ -325,7 +325,10 @@ module.exports = {
                  */
                 const otherAssignmentAfterTargetAssignment = allAssignments
                     .find(assignment => {
-                        if (assignment === targetAssignment || assignment.segment !== targetAssignment.segment) {
+                        if (
+                            assignment === targetAssignment ||
+                            assignment.segments.every(segment => !targetAssignment.segments.includes(segment))
```

Comment:
`every()` returns true when an array is empty. 
```suggestion
                            assignment.segments.length && assignment.segments.every(segment => !targetAssignment.segments.includes(segment))
```